### PR TITLE
Updated for 2.0.3 and some formatting fixes

### PR DESCRIPTION
**Pull request by qbzzt**

### CODE REVIEW

File: tutorials/integration.md [hunk 1]
```diff
@@ -345,122 +388,134 @@ for more information see here](https://medium.com/jeremy-keeshin/hello-world-for
...
-```
+&nsbp;
 
-Get the function to create a new configuration. This function provides the default 
-configuration, except for fields that are overridden by its parameter.
-```javascript
-const configureGSN =
-	require('@opengsn/gsn/dist/src/relayclient/GSNConfigurator').configureGSN
-```
+Define two variables that are singletons. By looking to see if there is a
```
Comment (forshtat) [inline tutorials/integration.md:456]: This sentence does not have a meaning. What do you mean?

File: tutorials/relay.md [hunk 2]
```diff
@@ -99,66 +96,47 @@ yes
...
-| versionRegistryAddress
-| The address for the version registry on the network you are using. 
-  [See this list](../deployments/networks.md).
-
-| ethereumNodeUrl
-| The URL to a node on the network you wish to use. If you do not know what to put here,
-  get a [free Infura account](https://infura.io), create a project, and look at 
-  **KEYS > ENDPOINTS** for your network. Use the endpoint that starts with https://
-|===
+<table>
```
Comment (drortirosh) [inline tutorials/relay.md:116]: don't wrtite html.
put "markdown" table (yes, the old table was adoc, which is different)

File: tutorials/relay.md [hunk 3]
```diff
@@ -201,20 +177,18 @@ docker run --rm -ti -v $PWD:$PWD \
     --relayUrl https://<your hostname for the relay>/gsn1 \
     -m `pwd`/pass12
 ```
-+
 {% hint style="note" %}
-### NOTE:
+##### NOTE:
```
Comment (drortirosh) [inline tutorials/relay.md:181]: I deliberately put it smaller, otherwise it looks HUGE on gitbook site (try to open it in its temporary site: docs2.opengsn.org

File: tutorials/integration.md [hunk 4]
```diff
@@ -150,31 +151,42 @@ import "@opengsn/gsn/contracts/forwarder/IForwarder.sol";
 import "@opengsn/gsn/contracts/BasePaymaster.sol";
 
 ```
+
```
Comment (drortirosh) [inline tutorials/integration.md:154]: remove manual formatting. look at the github generated site first, and we'll touch-up the template if needed, not "shotgun" the entire source with newlines.

File: tutorials/integration.md [hunk 5]
```diff
@@ -268,16 +311,21 @@ all the gas of the transaction, except for the unknown amount we are going to us
...
+&nbsp;
+
+This function returns the version of the GSN protocol the paymaster supports.
+In this case, the version is the latest at writing, `2.0.3`.
+
 ```solidity
     function versionPaymaster() external virtual view 
     override returns (string memory) {
-        return "1.0";
+        return "2.0.3";
```
Comment (drortirosh) [inline tutorials/integration.md:323]: great to fix this. this is a bug we had in our documentation, which broke the implementation of users...

File: tutorials/integration.md [hunk 6]
```diff
@@ -345,122 +388,134 @@ for more information see here](https://medium.com/jeremy-keeshin/hello-world-for
...
-+
 {% hint style="note" %}
-### NOTE:
+##### NOTE:
 At writing there is a bug that causes the output to have some junk characters. 
-{% endhint %}
 Under Linux you can use the `tr` command to solve this:
-+
 ```bash
 browserify index.js | tr -dc '\0-\177' > bundle.js
```
Comment (drortirosh) [inline tutorials/integration.md:416]: this actually works with 
`browserify index.js -o bundle.js`
with no mangling of character set.

File: tutorials/integration.md [hunk 7]
```diff
@@ -628,136 +708,93 @@ GSN started
...
-```
+1. Configure the settings you need to use GSN, similar to what you did in 
+   the user interface above, and create the `Provider` object.
+
+   ```javascript
+   Gsn = require("@opengsn/gsn/dist")
+   ethers = require("ethers")
+   origProvider = web3.currentProvider
+   conf = {verbose: false}
+   conf.forwarderAddress = '<address of the forwarder from gsn start>'
```
Comment (drortirosh) [inline tutorials/integration.md:739]: conf.forwarderAddress = require( '../build/gsn/Forwarder' ).address

File: tutorials/integration.md [hunk 8]
```diff
@@ -824,68 +856,89 @@ contract("CaptureTheFlag", async accounts => {
...
-		}
+
+&nbsp;
+
+The GSN provider is created the same way we did for manual testing, except for the 
+GSN parameters coming from `env` instead of being entered manually.
+
+```javascript
+                let gsnProvider = await
+                    new RelayProvider(web3provider, {
```
Comment (drortirosh) [inline tutorials/integration.md:914]: new 2.1 syntax:

`= await RelayProvider.newProvider( { provider: web3.provider, config: { } ).init()

File: tutorials/relay.md [hunk 9]
```diff
@@ -80,16 +80,13 @@ same GCP account, and that way be able to ssh to the relay VM.
...
 1. Set these parameters (accept the default for all the others):
-
-| Heading | Parameter | Value
-|---------|-----------|-------
-| Machine configuration | Machine type | e2-micro
-| Boot disk | Images | Debian GNU/Linux 10 (buster)
-| Identity and API access | Access scopes | All full access to all Cloud APIs
-
+<table>
+<tr><th>Heading</th><th>Parameter</th><th>Value</th></tr>
```
Comment (drortirosh) [inline tutorials/relay.md:84]: it was formatted as a tble. why did you change that?

File: tutorials/relay.md [hunk 10]
```diff
@@ -99,66 +96,47 @@ yes
 1. Delete the management VM, you no longer need it.
```
Comment (drortirosh) [inline tutorials/relay.md:96]: remove this "management VM". there is no such thing, and it was mistake to add.
It is a local unix machine (and I dont' care if the user decide to run it in a VM - locally or remote)

File: tutorials/integration.md [hunk 11]
```diff
@@ -345,122 +388,134 @@ for more information see here](https://medium.com/jeremy-keeshin/hello-world-for
...
+
 This is the configuration with the addresses of the relevant contracts (on the 
 test network where they are deployed, Kovan) and the maximum acceptable gas price for 
 our messages. The address of the contract we wish to contact is `conf.ourContract`.
 
 ```javascript
 const conf = {
-	ourContract: '0x23Cd0E36bB4727550bc01Cd3A1E8931b6d7CC796',
+	ourContract: '0x10A51A94d096e746E1Aec1027A0F8deCEC43FF63',
 	notOurs:     '0x6969Bc71C8f631f6ECE03CE16FdaBE51ae4d66B1',
```
Comment (forshtat) [inline tutorials/integration.md:446]: What is 'notOurs' and its role in this example?

File: tutorials/integration.md [hunk 12]
```diff
@@ -345,122 +388,134 @@ for more information see here](https://medium.com/jeremy-keeshin/hello-world-for
...
-	paymaster:   '0x0572dc46eb6edc950aa37c12fa9c862d4165cbc5',
-	relayhub:    '0x2E0d94754b348D208D64d52d78BcD443aFA9fa52',
-	stakemgr:    '0x0ecf783407C5C80D71CFEa37938C0b60BD255FF8',
+	paymaster:   '0x3f84367c25dC11A7aBE4B9ef97AB78d5D5498bF5',
+	relayhub:    '0xE9dcD2CccEcD77a92BA48933cb626e04214Edb92',
+        forwarder:   '0x0842Ad6B8cb64364761C7c170D0002CC56b1c498',
```
Comment (forshtat) [inline tutorials/integration.md:449]: I was under impression our new approach is to get these values (`relayHub` and `Forwarder`) form the Paymaster instance, is that correct? @drortirosh

File: tutorials/integration.md (hunk 11 above)
Comment (qbzzt) [reply tutorials/integration.md:446]: A contract that isn't our FlagCapture contract, so when we attempt to send a message to it the paymaster refuses to pay for it.

File: tutorials/integration.md [hunk 13]
```diff
@@ -214,39 +239,54 @@ Using this feature is beyond the scope of this tutorial.
...
-CAUTION: Advanced Paymaster implementations may choose to override the `getGasLimits()` method of the `IPaymaster` interface.
-Doing so can create to a configuration where paymaster commits to paying for a transaction after consuming some amount of gas.
-See the [Advanced](../gsn-provider/advanced.md) section to learn more.
+{% hint style="warning" %}
+##### CAUTION: 
+Advanced Paymaster implementations may choose to override the `getGasLimits()` 
+method of the `IPaymaster` interface.
+Doing so can create to a configuration where the paymaster commits to paying for a 
+transaction after consuming some amount of gas.
+{% endhint %}
```
Comment (forshtat) [inline tutorials/integration.md:264]: Do we not have an 'advanced' section anymore? Is there a reason the link to a place I assume some more info is given removed?

File: tutorials/integration.md [hunk 14]
```diff
@@ -345,122 +388,134 @@ for more information see here](https://medium.com/jeremy-keeshin/hello-world-for
...
-	relayHubAddress: conf.relayhub,
-	paymasterAddress: conf.paymaster,
-	stakeManagerAddress: conf.stakemgr,
-	gasPriceFactorPercent: 70,
-	methodSuffix: '_v4',
-	jsonStringifyRequest: true,
-```
+&nbsp;
+
+This function is asynchronous so it could be called mutiple times.
```
Comment (forshtat) [inline tutorials/integration.md:467]: There is no cause and effect relationship between function being asynchronous and there existing a possibility of it being called multiple times.

This entire paragraph basically describes to a reader the concept of a null check. This is outside the scope of documentation of any particular library and should be studied separately.

File: tutorials/integration.md [hunk 15]
```diff
@@ -345,122 +388,134 @@ for more information see here](https://medium.com/jeremy-keeshin/hello-world-for
...
+parameter to the `ethers` provider constructor.
 
-The `ethers.js` package uses its own provider class. So to create the provider we’ll use, 
-we take that GSN provider and use it as the parameter to the `ethers` provider constructor.
 ```javascript
-const origProvider = window.ethereum;
-const gsnProvider = new RelayProvider(origProvider, gsnConfig);
-const provider = new ethers.providers.Web3Provider(gsnProvider);
+
+        let gsnProvider = await new gsn.RelayProvider(window.ethereum, {
```
Comment (forshtat) [inline tutorials/integration.md:494]: Please change to `RelayProvider.newProvider` as of v. 2.1.0 (today :-) )

File: tutorials/integration.md [hunk 16]
```diff
@@ -345,122 +388,134 @@ for more information see here](https://medium.com/jeremy-keeshin/hello-world-for
...
+                paymasterAddress: conf.paymaster,
+                verbose: false}).init()
+	provider = new ethers.providers.Web3Provider(gsnProvider)
+	userAddr = gsnProvider.origProvider.selectedAddress
 ```
 
-Information about the contract we are accessing.
+&nbsp;
+
+For debugging purposes, it is useful to expose some variables in
```
Comment (forshtat) [inline tutorials/integration.md:504]: @drortirosh 
Is this a general-purpose advice we are willing to give at this stage of a tutorial? It does not seem GSN-specific to me.

File: tutorials/integration.md [hunk 17]
```diff
@@ -488,34 +543,40 @@ const flagAbi = [
...
+&nbsp;
+
 This function calls the contract using GSN. It is standard `ethers.js`, provided here for 
 completeness.
 ```javascript
 const gsnContractCall = async () => {
-
-	await window.ethereum.enable();
-
+	await startGsn()
```
Comment (forshtat) [inline tutorials/integration.md:552]: Why is `startGsn` here? First, it is confusing as we have `GsnTestEnvironment.startGsn` that literally starts the GSN on a local computer, which is not what this 'startGSN' does. 
Also, this code will only initialize the GSN-related variables once the user clicks or whatever, which is a weird timing. And will be called multiple times instead of once.


As the purpose of the tutorial is to showcase the library, the code must be dogmatic, trivial and immediately understandable. This does not feel like it.

File: tutorials/integration.md [hunk 18]
```diff
@@ -534,27 +595,37 @@ const gsnPaymasterRejection = async () => {
...
+        window.app.listenContract = contract
 ```
 
-This is an easy way to listen to events in `ethers.js`, and to parse them
-once they are received
+&nbsp;
+
+This is an easy way to listen to events in `ethers.js` version 5. When you
+listen for an event with n parameters, those values are the first n 
+parameters. The final parameter is the complete event information.
```
Comment (forshtat) [inline tutorials/integration.md:616]: This information is very useful but is not a documentation of GSN in my understanding of things.

File: tutorials/integration.md (hunk 12 above)
Comment (drortirosh) [reply tutorials/integration.md:449]: indeed. we're no longer configuring client with anything but paymaster.

File: tutorials/integration.md (hunk 1 above)
Comment (drortirosh) [reply tutorials/integration.md:456]: more specifically: we're not teaching anyone javascript, or event web3, for that matter.

File: tutorials/integration.md (hunk 18 above)
Comment (qbzzt) [reply tutorials/integration.md:616]: I wrote whatever I wished I had known when I started the project, to make things easier for implementers, because this is a tutorial, not a reference. If you insist I'll delete this and all information like it, but I think that would be a mistake.

File: tutorials/integration.md (hunk 8 above)
Comment (qbzzt) [reply tutorials/integration.md:914]: 1. When is 2.1 going to be released to npm?
2. Do we need to change the version in the contracts too, or does 2.1 apply only to the JavaScript?

Comment (forshtat) [reply tutorials/integration.md:914]: 1. It was just released a few hours ago
2. No, the contracts remain exactly the same for now

File: tutorials/integration.md (hunk 11 above)
Comment (forshtat) [reply tutorials/integration.md:446]: Are you sure this is clear from the immediate context of this line? Maybe it could be beneficial to showcase that in a different place?

File: tutorials/integration.md (hunk 18 above)
Comment (forshtat) [reply tutorials/integration.md:616]: This means your target audience is basically yourself from the past. I don't think this is a right assumption.

For those who already know the 'extra stuff' in your tutorial it will be hard to filter out the water.

For those who don't know it the given detail will be insufficient.

Large portions of this text do not relate to GSN and this is a missed opportunity to describe the system to those who are ready to learn about it in isolation from basic programming techniques, JavaScript syntax and specifics of libraries used in the tutorial.

File: tutorials/integration.md (hunk 4 above)
Comment (qbzzt) [reply tutorials/integration.md:154]: Do you mean the &nbsp; ? When I have code and paragraphs interspaced there's a risk that people get confused which code goes with which paragraph. I don't think that can be solved in the template.

File: tutorials/integration.md (hunk 6 above)
Comment (qbzzt) [reply tutorials/integration.md]: In some environments. I still get the error: 
image

so this needs to stay

File: tutorials/relay.md (hunk 2 above)
Comment (qbzzt) [reply tutorials/relay.md:116]: I tried a markdown table first. It didn't work with the formatting I wanted to put there.

File: tutorials/relay.md (hunk 9 above)
Comment (qbzzt) [reply tutorials/relay.md:84]: Because a markdown table resets the step numbering

image

File: tutorials/relay.md (hunk 2 above)
Comment (qbzzt) [reply tutorials/relay.md:116]: Markdown tables reset the ordered list I use for steps.

image

File: tutorials/integration.md (hunk 6 above)
Comment (qbzzt) [reply tutorials/integration.md:418]: I double checked, it still fails

image

When the step is done is exactly as you specified
```
#! /bin/bash

browserify etherless.js -o bundle.js
# | tr -dc '\0-\177' > bundle.js
mv bundle.js /var/www/html
cp etherless.html /var/www/html

echo Done at
date
```

Same result

File: tutorials/integration.md (hunk 7 above)
Comment (qbzzt) [reply tutorials/integration.md:592]: When I start GSN manually, it creates this file (`build/gsn/Forwarder.json`). But when I use the JavaScript `startGsn` , it does not get created.

File: tutorials/integration.md (hunk 8 above)
Comment (qbzzt) [reply tutorials/integration.md:767]: I get this error:
```
     TypeError: Cannot read property 'host' of undefined
````

I use this:
```
                       await RelayProvider.newProvider({
                                provider: web3.provider,
                                config: {
                                        forwarderAddress,
                                        paymasterAddress: paymaster.address
                                }
                        } ).init()
```

Can you tell me what I'm missing?

File: tutorials/relay.md (hunk 10 above)
Comment (qbzzt) [reply tutorials/relay.md:96]: Would a local Unix machine be authorized to ssh into the relay VM to change the configuration there?

**Comment by drortirosh** [comment]
Yes. When calling startGsn (), it returns the deployment, and doesn't
create any local file with addresses.
Btw, you can load the files by calling loadDeployment (), instead of
opening the files for each contact manually.

File: tutorials/relay.md (hunk 2 above)
Comment (qbzzt) [reply tutorials/relay.md:116]: Works property with Vuepress, so I can use md tables.

File: tutorials/integration.md [hunk 19]
```diff
@@ -343,213 +383,85 @@ for more information see here](https://medium.com/jeremy-keeshin/hello-world-for
...
+   ```bash
+   npm init -y
+   ```
 1. Install the GSN package and its dependencies:
-+
-```bash
-npm install @opengsn/gsn@"^2.0.0" ethers
-```
+   ```bash
+   npm install @opengsn/gsn@"^2.0.3" ethers
```
[the remark this answers is not in the document]
Comment (drortirosh) [reply tutorials/integration.md:397]: (you also have it in other place in this doc, but I can't comment there)

File: tutorials/integration.md [hunk 20]
```diff
@@ -343,213 +383,85 @@ for more information see here](https://medium.com/jeremy-keeshin/hello-world-for
...
+   ```
 1. Write your code in a file, for example `index.js`. You can use 
-[`require`](https://nodejs.org/en/knowledge/getting-started/what-is-require/) just as you 
-would with Node.js.
+   [`require`](https://nodejs.org/en/knowledge/getting-started/what-is-require/) just as you
```
Comment (drortirosh) [inline tutorials/integration.md:400]: please don't teach node. assume the user knows what require('module') means when creating a project.

File: tutorials/integration.md (hunk 6 above)
Comment (drortirosh) [reply tutorials/integration.md:412]: AFAIK, latest browserify works, but never mind about it.

File: tutorials/integration.md (hunk 12 above)
Comment (drortirosh) [reply tutorials/integration.md:440]: please read relayhub, forwarder from paymaster, and don't hard-code more addresses

File: tutorials/integration.md [hunk 21]
```diff
@@ -599,161 +491,121 @@ use a test network such as Kovan, but such a network is too slow for serious tes
...
+   truffle console
+   ```
 1. Deploy the paymaster as explained above.
-+
-[source.javascript]
-```
-paymaster = await <paymaster contract>.new()
-paymaster.setRelayHub(<Relay hub address from gsn command>)
-```
+   ```javascript
```
Comment (drortirosh) [inline tutorials/integration.md:527]: why deploy and configure paymaster here? if this part is about client, then use the paymaster deployed by  `gsn start` .
You should use this to get the addresses:
```
const { paymasterAddress, forwarderAddress } = 
    GsnTestEnvironment.loadDeployment()
```

File: tutorials/integration.md (hunk 7 above)
Comment (drortirosh) [reply tutorials/integration.md:550]: `loadDeployment()` is actually a better way to read the address than the `require( 'Forwarder').address`
calling `startGsn()` doesn't write the addresses to file, but returns them.
```
const { contractsDeployment } = await startGsn()
```
or:
```
const contractsDeployment = loadDeployment()
```
(don't write it, but currently our test env sucks and doesn't let you easily switch between the two options, so you need to select one and stick to it.)

File: tutorials/integration.md [hunk 22]
```diff
@@ -599,161 +491,121 @@ use a test network such as Kovan, but such a network is too slow for serious tes
...
+   the user interface above, and create the `Provider` object.
+
+   ```javascript
+   Gsn = require("@opengsn/gsn/dist")
+   ethers = require("ethers")
+   origProvider = web3.currentProvider
+   conf = {verbose: false}
+   conf.forwarderAddress = '<address of the forwarder from gsn start>'
+   conf.paymasterAddress = paymaster.address
+   gsnProvider = await new Gsn.RelayProvider(origProvider, conf).init()
```
Comment (drortirosh) [inline tutorials/integration.md:552]: use `newProvider`

File: tutorials/integration.md [hunk 23]
```diff
@@ -599,161 +491,121 @@ use a test network such as Kovan, but such a network is too slow for serious tes
...
-```
+   ```javascript
+   transaction = await contract.captureFlag()
+   receipt = await provider.waitForTransaction(transaction.hash)
+   ```
+1. Get the `_to` value from the receipt, and compare it with the 
+   account that signed the request.
+   ```javascript
+   receipt.logs.map(entry => contract.interface.parseLog(entry))[1].values._to
+   acct.address
```
Comment (drortirosh) [inline tutorials/integration.md:573]: broken line?

File: tutorials/integration.md [hunk 24]
```diff
@@ -792,26 +642,23 @@ events emitted by the transaction. To get the last caller from the event require
...
-	.
-	.
-	const result = receipt.logs.
-		map(entry => contract.interface.parseLog(entry)).
-		filter(entry => entry != null)[0];
-
-	return result.values['0']
-
+                const transaction = await contract.captureFlag()
+                const receipt = await provider.waitForTransaction(transaction.hash)
```
Comment (drortirosh) [inline tutorials/integration.md:648]: I think you can get the logs by calling await transaction.wait(), without this very cumbersome filtering.

File: tutorials/integration.md [hunk 25]
```diff
@@ -820,68 +667,89 @@ contract("CaptureTheFlag", async accounts => {
...
+
+```javascript
+                let env = await GsnTestEnvironment.startGsn('localhost')
 ```
 
+&nbsp;
+
 The various addresses GSN uses on the blockchain are stored in 
-`gsnInstance.deploymentResult`. Use them to initialize our contracts.
+`env.deploymentResult`. Use them to initialize our contracts.
```
Comment (drortirosh) [inline tutorials/integration.md:681]: was renamed to `env.contractsResult` in 2.1.0..

File: tutorials/integration.md (hunk 8 above)
Comment (drortirosh) [reply tutorials/integration.md:725]: web3.currentProvider

File: tutorials/relay.md [hunk 26]
```diff
@@ -33,10 +34,10 @@ For every transaction you relay you can expect to earn `baseRelayFee` (wei)
 plus `pctRelayFee`% of the cost of the gas for the transaction (in addition to being
 reimbursed for gas used).
 
-::: tip 
+{% hint style="info" %}
```
Comment (drortirosh) [inline tutorials/relay.md:37]: you restored the broken tip tags...

File: tutorials/relay.md (hunk 9 above)
Comment (drortirosh) [reply tutorials/relay.md:84]: markdown reset numbering only if the table starts at column 1.
if the table is indented (as it should) - numbering continues.
html tables in document are un-editable stuff..

File: tutorials/relay.md (hunk 10 above)
Comment (drortirosh) [reply tutorials/relay.md:96]: I don't care if the user uses a "management linux VM", or run it locally - but we're certainly don't SUGGEST to use such a temporary management box.
The doc can simply assume your local machine is linux (or mac) box , where all commands can easily be ran.
(the most we can do is suggest for windows user to use a VM locally)

File: tutorials/relay.md [hunk 27]
```diff
@@ -99,66 +96,47 @@ yes
...
 rdc logs -f https-portal
 ```
 1. When you see this line it means the setup is done. You can close the SSH window.
-+
 [source]
 ```
 [services.d](https-portal_1  | ) done.
 ```
-1. Browse to https://<your-DNS-name>/gsn1/getaddr. 
+1. Browse to https://&lt;your&nbsp;DNS&nbsp;name&gt;/gsn1/getaddr.
```
Comment (drortirosh) [inline tutorials/relay.md:139]: remove the "dot" at the end. it will break the URL

**Review by drortirosh — CHANGES_REQUESTED** [review]

File: tutorials/integration.md (hunk 7 above)
Comment (qbzzt) [reply tutorials/integration.md:550]: I would rather just show people both options, one for manual testing, the other for automated.

File: tutorials/relay.md (hunk 9 above)
Comment (qbzzt) [reply tutorials/relay.md:84]: Thank you. I'm fixing it.

File: tutorials/integration.md (hunk 11 above)
Comment (qbzzt) [reply tutorials/integration.md:437]: I just removed that function

File: tutorials/integration.md (hunk 17 above)
Comment (qbzzt) [reply tutorials/integration.md]: Renamed

File: tutorials/integration.md (hunk 19 above)
[the remark this answers is not in the document]
Comment (drortirosh) [reply tutorials/integration.md:397]: this is not resolved - its wrong version, and wrong quotes place (no need for quotes at all). if you DO need quotes, (e.g if windows doesn't accept `@opengsn/gsn^2.1.0` as a string) then put qutoes around entire string.

File: tutorials/integration.md (hunk 20 above)
Comment (drortirosh) [reply tutorials/integration.md:400]: remove this. we're not teaching first-grade javascript. normal app contains many `require`s, so it look ridiculous to explain it.

File: tutorials/integration.md (hunk 11 above)
Comment (drortirosh) [reply tutorials/integration.md:437]: lost you here. what function?

File: tutorials/integration.md (hunk 7 above)
Comment (drortirosh) [reply tutorials/integration.md:550]: both are automated: one is for starting GSN from within the test, and the other is for starting GSN just before the test.

File: tutorials/integration.md (hunk 13 above)
Comment (qbzzt) [reply tutorials/integration.md]: @drortirosh , do we have the advanced paymaster on the new site? I couldn't find it.
